pkg/prometheus/operator.go

```diff
@@ -19,17 +19,47 @@ import (
 	"reflect"
 	"regexp"
 	"strings"
+	"time"
 
 	monitoringv1 "github.com/prometheus-operator/prometheus-operator/pkg/apis/monitoring/v1"
+	monitoringv1alpha1 "github.com/prometheus-operator/prometheus-operator/pkg/apis/monitoring/v1alpha1"
+	monitoringclient "github.com/prometheus-operator/prometheus-operator/pkg/client/versioned"
+	"github.com/prometheus-operator/prometheus-operator/pkg/informers"
+	"github.com/prometheus-operator/prometheus-operator/pkg/k8sutil"
+	"github.com/prometheus-operator/prometheus-operator/pkg/operator"
 
 	"github.com/pkg/errors"
+	appsv1 "k8s.io/api/apps/v1"
 	v1 "k8s.io/api/core/v1"
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/fields"
+	"k8s.io/apimachinery/pkg/labels"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/rest"
+)
+
+const (
+	ResyncPeriod = 5 * time.Minute
 )
 
 var prometheusKeyInShardStatefulSet = regexp.MustCompile("^(.+)/prometheus-(.+)-shard-[1-9][0-9]*$")
 var prometheusKeyInStatefulSet = regexp.MustCompile("^(.+)/prometheus-(.+)$")
 
+type CommonConfig struct {
+	ClusterConfig *rest.Config
+	KClient       *kubernetes.Clientset
+	MClient       *monitoringclient.Clientset
```

Comment:
I'm not sure about the benefit of having these fields. The struct feels a bit like a mixed bag of unrelated things and I'd focus on informers first.

Reply:
But those clients needed to initialize the informers, if not moving them here..
We will need to duplicate them (init them twice).
I don't think this is a good idea.

Reply:
I agree with @simonpasquier that I'm feeling a trend with this and this [1], that we will end up with multiple structs that will have the clients duplicated. Now I'm not sure what @simonpasquier has in mind, but maybe we could have a struct called `Clients` that would initialise and pass to the Operators which would then, in turn, pass them to functions like `Process` or `InitCommonConfig`. But let's see what @simonpasquier suggests

[1] https://github.com/prometheus-operator/prometheus-operator/pull/5493

Reply:
I'm not sure, if I understand this well.